### PR TITLE
[MASTER] - Documentação sobre prop spreading em componentes wrapper

### PR DESCRIPTION
**Pull request by luiz-ortega**
# Add spread operator in react wrapper components with typescript:

### 7.7 Using spread operator

When creating a component wrapper we can spread the types from our original component. That way the wrapper extends all the props from the original component automatically. Creating a custom type and passing on `React.FC` typescript can give us all the correct props. This is usefull to avoid creating a custom interface for our wrapper with missing props from the original component.

**✅ Good:**

```tsx
import { MenuItem, TextField } from '@mui/material';
import { TextFieldProps } from '@mui/material';

export type SelectOption = { value: string; label: string };

export type SelectProps = TextFieldProps & {
  options: SelectOption[];
};

const Select: React.FC<SelectProps> = ({ options, ...props }) => {
  return (
    <TextField {...props}>
      {options.map((option) => (
        <MenuItem key={uuidv4()} value={option.value}>
          {option.label}
        </MenuItem>
      ))}
    </TextField>
  );
};
```

**❌ Bad:**

```tsx
import { MenuItem, TextField } from '@mui/material';

export type SelectOption = { value: string; label: string };

export type SelectProps = {
  options: SelectOption[];
  disabled: boolean;
  onChange: () => void;
  value: string;
  onBlur: () => void;
};

const Select: React.FC<SelectProps> = ({
  options,
  disabled,
  onChange,
  value,
  onBlur,
}) => {
  return (
    <TextField
      disabled={disabled}
      onChange={handleOnChange}
      value={value}
      onBlur={handleOnBlur}
    >
      {options.map((option) => (
        <MenuItem key={uuidv4()} value={option.value}>
          {option.label}
        </MenuItem>
      ))}
    </TextField>
  );
};
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1229,14 +1279,14 @@ To define the `test-id` to a component use the follow structure: `[page-name||co
 
 **✅ Good:**
 
-- forgot-password__input--email
-- header__select--cnpjList
-- login__button--forgot-password
+- forgot-password\_\_input--email
```
Comment (RodrigoRVSN) [inline README.md:1282]: question: why this change?

**Comment by RodrigoRVSN** [comment]
@luiz-ortega add in PR description whats being changed

**Review by felipefialho — CHANGES_REQUESTED** [review]
This PR completely changed the document, I suggest reverting to the alteration and doing it again

File: README.md (hunk 1 above)
Comment (luiz-ortega) [reply README.md]: VSCode auto format fool me again :disappointed: 
I already revert it.

Comment (RodrigoRVSN) [reply README.md]: Now the PR is empty @luiz-ortega 

image

Comment (luiz-ortega) [reply README.md]: @RodrigoRVSN yes, i had it reverted so i could readded all the alterations. It's solved now!

File: README.md [hunk 2]
```diff
@@ -1034,6 +1034,75 @@ const MyComponent = (type) => {
 }
 ```
 
+### 7.8 Using spread operator
+
+When creating a component wrapper we can spread the types from our original component. That way the wrapper extends all the props from the original component automatically. Creating a custom type and passing on `React.FC` typescript can give us all the correct props. This is usefull to avoid creating a custom interface for our wrapper with missing props from the original component.
```
Comment (RodrigoRVSN) [inline README.md:1039]: typo: 
```suggestion
When creating a component wrapper we can spread the types from our original component. That way the wrapper extends all the props from the original component automatically. Creating a custom type and passing on `React.FC` typescript can give us all the correct props. This is useful to avoid creating a custom interface for our wrapper with missing props from the original component.
```

File: README.md [hunk 3]
```diff
@@ -1034,6 +1034,75 @@ const MyComponent = (type) => {
...
+import { MenuItem, TextField } from '@mui/material';
+import { TextFieldProps } from '@mui/material';
+
+export type SelectOption = { value: string; label: string };
+
+export type SelectProps = TextFieldProps & {
+  options: SelectOption[];
+};
+
+const Select: React.FC<SelectProps> = ({ options, ...props }) => {
```
Comment (mauriciomutte) [inline README.md:1053]: I think using React.FC to define prop types should not be encouraged. This is a old practice that I don't see anymore in the community (see a [discussion](https://github.com/facebook/create-react-app/pull/8177) in CRA project). 

Instead we could define type direct in prop argument: `MyComponent(props: MyComponentProps)`. Another example:

```suggestion
const Select = ({ options, ...props }: SelectProps) => {
```

**Review by RodrigoRVSN — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -1034,6 +1034,75 @@ const MyComponent = (type) => {
 }
 ```
 
+### 7.8 Using spread operator
+
+When creating a component wrapper we can spread the types from our original component. That way the wrapper extends all the props from the original component automatically. Creating a custom type and passing on `React.FC` typescript can give us all the correct props. This is useful to avoid creating a custom interface for our wrapper with missing props from the original component.
```
Comment (mauriciomutte) [inline README.md:1039]: React.FC was forgotten here 😅

```suggestion
When creating a component wrapper we can spread the types from our original component. That way the wrapper extends all the props from the original component automatically. Creating a custom type typescript can give us all the correct props. This is useful to avoid creating a custom interface for our wrapper with missing props from the original component.
```

**Review by mauriciomutte — DISMISSED** [review]

**Review by felipefialho — APPROVED** [review]

**Review by RodrigoRVSN — APPROVED** [review]

**Review by mauriciomutte — APPROVED** [review]
